packages/safe-tools-client/app/components/network-connect-modal/index.gts

```diff
@@ -130,17 +91,17 @@ class NetworkConnectModal extends Component<Signature> {
               @orientation="horizontal"
               @spacing="default"
               @groupDescription='Select a wallet to connect to'
-              @items={{this.walletProviders}}
-              @disabled={{@isConnecting}}
-              @checkedId={{this.walletProviderId}}
+              @items={{this.wallet.walletProviders}}
+              @disabled={{this.wallet.isConnecting}}
+              @checkedId={{this.wallet.providerId}}
               @hideRadio={{true}}
               class='network-connect-modal__wallet-group'
               data-test-wallet-selection as |option|
             >
               {{#let option.data as |item|}}
                 <option.component
                   @name='wallet-provider-selection'
-                  @onChange={{fn (mut this.walletProviderId) item.id}}
+                  @onChange={{fn (mut this.wallet.providerId) item.id}}
```

Comment:
Probably doesn't need to be in this PR, but I think `providerId` is trying to do two things at once, and they should be separated:

1. It's representing the UI state of the radio buttons. This should live in this component, close to the UI, and get passed to `WalletService.connect` as an argument. 

2. It's telling us which wallet provider is currently connected. This should be derived from a property of `ChainConnectionManager` (We probably should have done this in layer 1 chain in web-client, maybe we could make `strategy` a tracked property?).

Reply:
I’ve added a property local to the component, thanks. I think the direction of #2 makes sense and I’ll try to include it in forthcoming work 😀